packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -241,3 +248,310 @@ export class Load extends Op {
     stack.push(this.interpreter.scratch[this.index]);
   }
 }
+
+// err opcode : Error. Panic immediately.
+export class Err extends Op {
+  execute (stack: TEALStack): void {
+    throw new TealError(ERRORS.TEAL.TEAL_ENCOUNTERED_ERR);
+  }
+}
+
+// SHA256 hash of value X, yields [32]byte
+export class Sha256 extends Op {
+  execute (stack: TEALStack): void {
+    const hash = sha256.create();
+    const val = this.assertBytes(stack.pop()) as Message;
+    hash.update(val);
+    const hashedOutput = Buffer.from(hash.hex(), 'hex');
+    var arrByte = Uint8Array.from(hashedOutput);
+    stack.push(arrByte);
+  }
```

Comment:
Question: Implementation of `Sha256`, `Itob`, `Btoi` seems general rather than specific to some opcodes. Do you think it's possible to create general functions (eg - `Sha256`, `intToBytes(..)`, ..) in `/lib` for them and use those functions directly here?

Reply:
can you please elaborate ? what is general and specific in this context ?

Reply:
I mean we can use `Sha256(data_to_encode)`,.. in other places as well. Right?

Reply:
yes, we can import library and use them anywhere.

Reply:
ok cool!